### PR TITLE
DEV: simplify optin controller

### PR DESCRIPTION
**Pull request by TonisOrmisson**
Fixed issue # : 
New feature # :
Changed feature # :
Dev: 
Dev:

### CODE REVIEW

File: application/controllers/OptinController.php [hunk 1]
```diff
@@ -42,7 +42,7 @@ function actiontokens($surveyid, $token, $langcode = '')
         if (!$iSurveyID) {
             $this->redirect(array('/'));
         }
-        $iSurveyID = (int) $iSurveyID;
+        $iSurveyID = $oSurvey->primaryKey;
```
Comment (Shnoulle) [inline application/controllers/OptinController.php]: Maybe can remove this line : https://github.com/TonisOrmisson/LimeSurvey/blob/b8af9b7256deb22e8ed8d8b3b7f67b86652182cc/application/controllers/OptinController.php#L37 (directly use  `$oSurvey = Survey::model()->findByPk($surveyid);` )

And send a 404 error if $oSurvey is null

File: application/controllers/OptinController.php [hunk 2]
```diff
@@ -54,13 +54,11 @@ function actiontokens($surveyid, $token, $langcode = '')
 
         Yii::app()->setLanguage($sBaseLanguage);
 
-        $aSurveyInfo = getSurveyInfo($iSurveyID, $sBaseLanguage);
-
-        if ($aSurveyInfo == false || !tableExists("{{tokens_{$iSurveyID}}}")) {
+        if (empty($oSurvey) || !$oSurvey->hasTokensTable) {
```
Comment (Shnoulle) [inline application/controllers/OptinController.php]: empty test is not needed , done at https://github.com/TonisOrmisson/LimeSurvey/blob/b8af9b7256deb22e8ed8d8b3b7f67b86652182cc/application/controllers/OptinController.php#L42

File: application/controllers/OptinController.php [hunk 3]
```diff
@@ -54,13 +54,11 @@ function actiontokens($surveyid, $token, $langcode = '')
...
             throw new CHttpException(404, "This survey does not seem to exist. It may have been deleted or the link you were given is outdated or incorrect.");
         } else {
-            LimeExpressionManager::singleton()->loadTokenInformation($iSurveyID, $token, false);
-            $oToken = Token::model($iSurveyID)->findByAttributes(array('token' => $token));
+            LimeExpressionManager::singleton()->loadTokenInformation($iSurveyID, $sToken, false);
+            $oToken = Token::model($iSurveyID)->findByAttributes(array('token' => $sToken));
 
             if (!isset($oToken)) {
                 $sMessage = gT('You are not a participant in this survey.');
```
Comment (Shnoulle) [inline application/controllers/OptinController.php]: I like to have a HTML error here, unsure of code, maybe 403 ?

Comment (TonisOrmisson) [reply application/controllers/OptinController.php]: Maybe, but I was not looking at that here at all :)
Just there was a sanitized $sToken there that was not used and an unsanitized $token used instead

**Comment by Shnoulle** [comment]
Yep, but more : best to use 40X error when we have (again) a real clean error system

**Pull request merged**